src/Strategy/AbstractStrategy.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Http\Discovery\Strategy;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+abstract class AbstractStrategy implements DiscoveryStrategy
+{
+    protected static $classes = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function find($type)
+    {
+        if (isset(static::$classes[$type])) {
+            return static::$classes[$type];
```

Comment:
should this check if the class indeed exists? otherwise the concrete strategies would need to do that check.